File: PDP/DrivingSchoolAppAndroidClient/app/src/main/java/com/example/drivingschoolappandroidclient/api/models/GradeByInstructorToStudentModel.kt
package com.example.drivingschoolappandroidclient.api.models

/**
Модель для передачи данных оценки на сервер
 */
open class GradeByInstructorToStudentModel(
    var classId: Int,
    var grade: Byte,
    var comment: String?,
){
    var gradeEnum : GradesByInstructorsToStudents
        get() = GradesByInstructorsToStudents.ofValue(grade)
        set(value) {
            grade = value.value
        }
}

File: PDP/DrivingSchoolWebAPI3/DrivingSchoolAPIModels/ApiModels/GradeByInstructorToStudentModel.cs
using System.ComponentModel.DataAnnotations;

namespace DrivingSchoolAPIModels
{
    /// <summary>
    /// Модель для передачи данных оценки на сервер
    /// </summary>
    public class GradeByInstructorToStudentModel
    {
        [Key]
        public int ClassId { get; set; }
        public GradesByInstructorsToStudents Grade { get; set; }
        public string? Comment { get; set; }
    }
}
